datumaro/components/operations.py

```diff
@@ -145,14 +149,44 @@ def _merge_attrs(a: Dict[str, Any], b: Dict[str, Any],
 
         return merged
 
-    @staticmethod
-    def _merge_images(item_a: DatasetItem, item_b: DatasetItem) -> Image:
-        image = None
+    @classmethod
+    def _merge_media(cls, item_a: DatasetItem, item_b: DatasetItem) \
+            -> Union[ByteImage, Image, PointCloud, Video, VideoFrame]:
+        if isinstance(item_a.media, (ByteImage, Image, VideoFrame)) or \
+                isinstance(item_b.media, (ByteImage, Image, VideoFrame)):
+            media = cls._merge_images(item_a, item_b)
+        else:
```

Comment:
I think, it will be better to have a separate block for each case, even if it produces some code duplication.

Reply:
Divided into blocks by type